packages/roosterjs-content-model/lib/publicApi/utils/formatSegmentWithContentModel.ts

```diff
@@ -21,8 +21,15 @@ export function formatSegmentWithContentModel(
         editor,
         apiName,
         model => {
-            const segments = getSelectedSegments(model, !!includingFormatHolder);
+            let segments = getSelectedSegments(model, !!includingFormatHolder);
             const pendingFormat = editor.getPendingFormat();
+            const isCollapsedSelection =
+                segments.length == 1 && segments[0].segmentType == 'SelectionMarker';
+
+            if (isCollapsedSelection) {
+                segments = selectWord([model], segments[0]);
```

Comment:
[model] does not always work. When selection is under a list, or a table, or a quote, "path" parameter need to contain the whole block group path. However, we don't have this info here. So suggest we just pass the whole model object here, then let the function selectWord() to find the right start point.